tests/Elasticsearch/Metadata/Resource/Factory/ElasticsearchProviderResourceMetadataCollectionFactoryTest.php

```diff
@@ -80,10 +82,12 @@ public function testCreate(?bool $elasticsearchFlag, int $expectedCatCallCount,
                 'pri.store.size' => '42kb',
             ]]);
         } else {
-            $catNamespace->indices(['index' => 'foo'])->willThrow(new Missing404Exception());
+            $catNamespace->indices(['index' => 'foo'])->willThrow(new ClientResponseException());
         }
 
-        $client = $this->prophesize(Client::class);
+        $clientClass = class_exists(\Elasticsearch\Client::class) ? \Elasticsearch\Client::class : \Elastic\Elasticsearch\ClientInterface::class;
+
+        $client = $this->prophesize($clientClass);
```

Comment:
`Elastic\Elasticsearch\ClientInterface` doesn't have `cat` method defined inside of it and I don't have a clue how to mock method which doesn't exist with a help of **prophecy**. On other hand I couldn't mock `Elastic\Elasticsearch\Client` as it is marked as **final**. 

I see only one way here: write decorator for a ES Client on Api-Platform side

Reply:
We should avoid changing existing tests, duplicate the tests you need, don't test the `cat` with es > 8